node/caching-compiler.js

```diff
@@ -103,16 +111,18 @@ function readFileOrNull(filename) {
   }
 }
 
-function getCacheFilename(content, reify) {
-  const hash = createHash("sha1")
-    .update(reifyVersion).update("\0")
-    .update(content).update("\0");
-
-  if (reify) {
-    hash.update(JSON.stringify(reify));
+function getCacheFilename() {
+  const args = [];
+  const argc = arguments.length;
+  for (let i = 0; i < argc; ++i) {
+    args.push(arguments[i]);
   }
 
-  return hash.digest("hex") + ".js";
+  return createHash("sha1")
+    .update(reifyVersion)
+    .update("\0")
+    .update(JSON.stringify(args))
```

Comment:
There are some valid JS `content` values which are not valid inside string literals (like one or two bizarro encodings). That makes me a little worried about using `JSON.stringify` here. It also seems super heavy since we know the arg count is 2 and type detection for one of cacheKey or content is pretty straight forward.